apps/gdalwarp_lib.cpp

```diff
@@ -2502,10 +2502,21 @@ static GDALDatasetH GDALWarpDirect(const char *pszDest, GDALDatasetH hDstDS,
             CPLSPrintf("%g", psOptions->dfErrorThreshold));
     }
 
+    // Examine desired overview level and retrieve the corresponding dataset
+    // if it exists.
+    GDALDatasetH hDstDSOverview = nullptr;
+    if (psOptions->nOvLevel >= 0)
+    {
+        hDstDSOverview = GDALCreateOverviewDataset(
```

Comment:
This will leak hDstDSOverview, and more annoyingly this will crash if nSrcCount > 1.
I believe the call to GDALCreateOverviewDataset() should be moved to CreateOutput() and applied to all input datasets

Reply:
I see, it's much more complex that I imagined. 

I think I  have now the proper implementation but there is still a failing test that I don't know how to handle (what is the expected behavior):

```python
    # Crates a full res output file
    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif}")
    # The output file has now a "wrong" resolution and no overwrite was specified
    gdaltest.runexternal(f"{gdalwarp_path} {src_tif} {dst_tif} -ovr 0")
    ds = gdal.Open(dst_tif)
    # Fail! Full res is used instead of overview
    assert ds.GetRasterBand(1).Checksum() == cs_ov0
```

What are we supposed to do here?
Error out or silently overwrite the output file (that has full res instead of the ovr res)?

The code path in this case skips createOutput completely.

Reply:
this is also unexpected (even if the output res is correct) https://github.com/OSGeo/gdal/pull/10193/files#diff-143c752f11c4a0dd7cde1c002e04f4df6f174faebddab77e602b5afff6dfe8f8R1139

Reply:
> What are we supposed to do here?
> Error out or silently overwrite the output file (that has full res instead of the ovr res)?

There should already be a warning emitted at line 2860 (in master): "cannot get overview level ...". This is probably enough for now